sql-parser/src/main/antlr/SqlBase.g4

```diff
@@ -494,13 +494,13 @@ blobClusteredInto
     ;
 
 tableElement
-    : columnDefinition                                                               #columndDefinitionDefault
+    : columnDefinition                                                               #columnDefinitionDefault
     | PRIMARY_KEY columns                                                            #primaryKeyConstraint
     | INDEX name=ident USING method=ident columns withProperties?                    #indexDefinition
     ;
 
 columnDefinition
-    : ident dataType? ((GENERATED ALWAYS)? AS expr)? columnConstraint*
+    : ident dataType? (DEFAULT defaultExpr=expr)? ((GENERATED ALWAYS)? AS generatedExpr=expr)? columnConstraint*
```

Comment:
maybe we indeed have to enforce mutual elusiveness on the parser level? e.g.

```
columnDefinition
    : ident dataType? columnDefinitionExpressions columnConstraint*
    ;

columnDefinitionExpressions
    : (DEFAULT defaultExpr=expr)?
    | ((GENERATED ALWAYS)? AS generatedExpr=expr)?
    ;
```

Reply:
As discussed, the data type is also optional so there would still be the need to do some validation later on or add this in the parsing logic and capture all possible combinations (which does not worth the complexity imho).

Thus, I decided to keep the parser simple, do all validations in a single place and a huge plus for me is the (far more) descriptive user error messages.